doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+        close( unit )
+        open( newunit=unit, file='test.bin', status='old', &
+            form='unformatted', action='read' )
+        call set5 % input(unit)
+        call set4 % input(unit)
+        call set3 % input(unit)
+        close( unit )
+        if ( set3 /= set0 .or. set4 /= set1 .or. set5 /= set2 ) then
+            error stop 'Transfer to and from units using ' // &
+                ' output and input failed.'
+        else
+            write(*,*) 'Transfer to and from units using ' // &
+                'output and input succeeded.'
+        end if
+    end program demo_input
+```
+
+### `none` - determines whether no bits are set
+
+#### Status
+
+Experimental
+
+#### Description
+
+Determines whether no bits are set in `self`.
+
+#### Syntax
+
+`result = self % [[bitset_type(class):none(bound)]] ()`
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+`self`: shall be a scalar expression of class `bitset_type`. It is an
+  `intent(in)` argument.
+
+#### Result character
+
+The result is a default logical scalar.
+
+#### Result value
+
+The result is `.true.` if no bits in `self` are set, otherwise it is
+`.false.`.
+
+#### Example
+
+```fortran
+    program demo_none
+        use stdlib_bitsets
+        character(*), parameter :: &
+            bits_0 = '0000000000000000000'
+        type(bitset_large) :: set0
+        call set0 % from_string( bits_0 )
+        if ( set0 % none() ) then
+            write(*,*) "FROM_STRING interpreted " // &
+                "BITS_0's value properly."
+        end if
+        call set0 % set(5)
+        if ( .not. set0 % none() ) then
+            write(*,*) "NONE interpreted SET0's value properly."
+        end if
+    end program demo_none
+```
+
+### `not` - Performs the logical complement on a bitset
+
+#### Status
+
+Experimental
+
+#### Description
+
+Performs the logical complement on the bits of `self`.
+
+#### Syntax
+
+`result = self % [[bitset_type(class):not(bound)]] ()`
+
+#### Class
+
+Elemental subroutine.
+
+#### Argument
+
+`self` shall be a scalar variable of class `bitset_type`. It is an
+  `intent(inout)` argument. On return its bits shall be the logical
+  complement of their values on input.
+
+#### Example
+
+```fortran
+    program demo_not
+        use stdlib_bitsets
+        type(bitset_large) :: set0
+        call set0 % init( 155 )
+        if ( set0 % none() ) then
+            write(*,*) "FROM_STRING interpreted " // &
+                "BITS_0's value properly."
+        end if
+        call set0 % not()
+        if ( set0 % all() ) then
+            write(*,*) "ALL interpreted SET0's value properly."
+        end if
+    end program demo_not
+```
+
+### `or` - Bitwise OR of the bits of two bitsets.
+
+#### Status
+
+Experimental
+
+#### Description
+
+Replaces the original bits of `set1` with the bitwise `or` of those
+bits with the bits of `set2`. Note `set1` and `set2` must have the
+samee number of bits, otherwise the result is undefined.
```

Comment:
How is `undefined` defined? `NaN? (I didn't look to the code yet, but I think it would be good to define "undefined" in the specs.)